File: app/src/main/java/com/zillennium/utswap/screens/setting/MyWallet/dialog/ExportTransactionBottomSheetDialog.java
package com.zillennium.utswap.screens.setting.MyWallet.dialog;

import android.app.DatePickerDialog;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
import com.google.android.material.button.MaterialButton;
import com.zillennium.utswap.R;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Locale;
import java.util.Objects;

public class ExportTransactionBottomSheetDialog extends BottomSheetDialogFragment implements AdapterView.OnItemSelectedListener {

    private View view;
    private EditText etStart, etEnd;
    private MaterialButton btnClear, btnExport;
    private Spinner spinner;
    private final String[] strPeriod = {"Select Time","Today", "Yesterday", "This week", "Last week", "This Month", "Last Month"};

    public static ExportTransactionBottomSheetDialog newInstance(){
        ExportTransactionBottomSheetDialog exportTransactionBottomSheetDialog = new ExportTransactionBottomSheetDialog();
        exportTransactionBottomSheetDialog.setArguments(new Bundle());
        return exportTransactionBottomSheetDialog;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        view =inflater.inflate(R.layout.layout_export_transaction_bottomsheet_fragment,container,false);

        Objects.requireNonNull(getDialog()).requestWindowFeature(Window.FEATURE_NO_TITLE);
        getDialog().getWindow().setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));

        initView();

        spinner.setOnItemSelectedListener(this);
        ArrayAdapter<String> aa = new ArrayAdapter<String>(this.getContext(), android.R.layout.simple_spinner_item,strPeriod);
        aa.setDropDownViewResource(android.R.layout
                .simple_spinner_dropdown_item);
        spinner.setAdapter(aa);

        initAction();

        return view;
    }

    public void initView(){
        etStart = view.findViewById(R.id.et_start);
        etEnd = view.findViewById(R.id.et_end);
        btnClear = view.findViewById(R.id.btn_clear);
        btnExport = view.findViewById(R.id.btn_export);
        spinner = view.findViewById(R.id.spinner_period);
    }

    public void initAction(){
        btnExport.setOnClickListener(view1 -> {
            dismiss();
        });

        btnClear.setOnClickListener(view1 -> {
            dismiss();
        });

        Calendar calendar = Calendar.getInstance();

        DatePickerDialog.OnDateSetListener date_start = (datePicker, year, month, day) -> {

            calendar.set(Calendar.YEAR, year);
            calendar.set(Calendar.MONDAY, month);
            calendar.set(Calendar.DAY_OF_MONTH, day);

            String Format = "MM/dd/yyyy";
            SimpleDateFormat simpleDateFormat = new SimpleDateFormat(Format, Locale.US);

            etStart.setText(simpleDateFormat.format(calendar.getTime()));

        };

        DatePickerDialog.OnDateSetListener date_end = (datePicker, year, month, day) -> {

            calendar.set(Calendar.YEAR, year);
            calendar.set(Calendar.MONDAY, month);
            calendar.set(Calendar.DAY_OF_MONTH, day);

            String Format = "MM/dd/yyyy";
            SimpleDateFormat simpleDateFormat = new SimpleDateFormat(Format, Locale.US);

            etEnd.setText(simpleDateFormat.format(calendar.getTime()));

        };

        etStart.setOnClickListener(view -> new DatePickerDialog(this.getContext(), date_start, calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH), calendar.get(Calendar.DAY_OF_MONTH)).show());

        etEnd.setOnClickListener(view -> new DatePickerDialog(this.getContext(), date_end, calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH), calendar.get(Calendar.DAY_OF_MONTH)).show());

        if(spinner.getSelectedItem().toString().equals("Today"))
        {
//            etStart.setText(Utils.getTodayDate());
//            etEnd.setText(Utils.getTodayDate());
        }
    }

    @Override
    public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
        Toast.makeText(this.getContext(),strPeriod[i] , Toast.LENGTH_LONG).show();
    }

    @Override
    public void onNothingSelected(AdapterView<?> adapterView) {

    }
}

File: app/src/main/java/com/zillennium/utswap/screens/wallet/MyWallet/dialog/ExportTransactionBottomSheetDialog.kt
package com.zillennium.utswap.screens.wallet.MyWallet.dialog

import android.app.DatePickerDialog
import android.app.DatePickerDialog.OnDateSetListener
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.Window
import android.widget.*
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import com.google.android.material.button.MaterialButton
import com.zillennium.utswap.R
import com.zillennium.utswap.UTSwapApp
import java.text.SimpleDateFormat
import java.util.*


class ExportTransactionBottomSheetDialog : BottomSheetDialogFragment(),
    AdapterView.OnItemSelectedListener {
    internal var view: View? = null
    private var etStart: EditText? = null
    private var etEnd: EditText? = null
    private var btnClear: MaterialButton? = null
    private var btnExport: MaterialButton? = null
    private var spinner: Spinner? = null
    private val strPeriod = arrayOf(
        "Select Time",
        "Today",
        "Yesterday",
        "This week",
        "Last week",
        "This Month",
        "Last Month"
    )

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        view = inflater.inflate(
            R.layout.layout_export_transaction_bottomsheet_fragment,
            container,
            false
        )
        Objects.requireNonNull(dialog)?.requestWindowFeature(Window.FEATURE_NO_TITLE)
        dialog!!.window!!.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
        initView()
        spinner!!.onItemSelectedListener = this
        val aa = ArrayAdapter(UTSwapApp.instance, android.R.layout.simple_spinner_item, strPeriod)
        aa.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        spinner!!.adapter = aa
        initAction()
        return view
    }

    fun initView() {
        etStart = view?.findViewById(R.id.et_start)
        etEnd = view?.findViewById(R.id.et_end)
        btnClear = view?.findViewById(R.id.btn_clear)
        btnExport = view?.findViewById(R.id.btn_export)
        spinner = view?.findViewById(R.id.spinner_period)
    }

    fun initAction() {
        btnExport!!.setOnClickListener { dismiss() }
        btnClear!!.setOnClickListener { dismiss() }
        val calendar = Calendar.getInstance()
        val dateStart =
            OnDateSetListener { _: DatePicker?, year: Int, month: Int, day: Int ->
                calendar[Calendar.YEAR] = year
                calendar[Calendar.MONDAY] = month
                calendar[Calendar.DAY_OF_MONTH] = day
                val format = "MM/dd/yyyy"
                val simpleDateFormat =
                    SimpleDateFormat(format, Locale.US)
                etStart!!.setText(simpleDateFormat.format(calendar.time))
            }
        val dateEnd =
            OnDateSetListener { _: DatePicker?, year: Int, month: Int, day: Int ->
                calendar[Calendar.YEAR] = year
                calendar[Calendar.MONDAY] = month
                calendar[Calendar.DAY_OF_MONTH] = day
                val format = "MM/dd/yyyy"
                val simpleDateFormat =
                    SimpleDateFormat(format, Locale.US)
                etEnd!!.setText(simpleDateFormat.format(calendar.time))
            }
        etStart!!.setOnClickListener {
            DatePickerDialog(
                UTSwapApp.instance,
                dateStart,
                calendar[Calendar.YEAR],
                calendar[Calendar.MONTH],
                calendar[Calendar.DAY_OF_MONTH]
            ).show()
        }
        etEnd!!.setOnClickListener {
            DatePickerDialog(
                UTSwapApp.instance,
                dateEnd,
                calendar[Calendar.YEAR],
                calendar[Calendar.MONTH],
                calendar[Calendar.DAY_OF_MONTH]
            ).show()
        }
    }

    override fun onItemSelected(adapterView: AdapterView<*>?, view: View, i: Int, l: Long) {
        Toast.makeText(UTSwapApp.instance, strPeriod[i], Toast.LENGTH_LONG).show()
    }

    override fun onNothingSelected(adapterView: AdapterView<*>?) {}

    companion object {
        fun newInstance(): ExportTransactionBottomSheetDialog {
            val exportTransactionBottomSheetDialog = ExportTransactionBottomSheetDialog()
            exportTransactionBottomSheetDialog.arguments = Bundle()
            return exportTransactionBottomSheetDialog
        }
    }
}